# Questions of interest
* How many categories of aircraft are there in the dataset? <br>
* How many accidents and incidents are there? <br>
* How many accidents and incidents are there by category? <br>

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt 

In [3]:
df = pd.read_csv("AviationData.csv",encoding = "ISO-8859-1",low_memory = False, parse_dates = ["Event.Date"],usecols = ["Investigation.Type","Accident.Number","Event.Date","Aircraft.damage","Aircraft.Category","Number.of.Engines","Engine.Type","Air.carrier","Total.Fatal.Injuries"])

In [24]:
df.tail()

Investigation.Type Accident.Number Event.Date Aircraft.damage  \
88536           Accident      ANC22LA077 2022-09-16     Substantial   
88549           Accident      ANC22LA078 2022-09-19     Substantial   
88565           Accident      ANC22LA081 2022-09-23     Substantial   
88600           Accident      CEN22LA441 2022-09-29     Substantial   
88615           Accident      CEN23LA002 2022-10-02     Substantial   

      Aircraft.Category  Number.of.Engines    Engine.Type  \
88536          Airplane                1.0  Reciprocating   
88549          Airplane                1.0  Reciprocating   
88565          Airplane                1.0  Reciprocating   
88600          Airplane                1.0  Reciprocating   
88615          Airplane                1.0  Reciprocating   

                         Air.carrier  Total.Fatal.Injuries  Years  
88536                       Acme Cub                   0.0   2022  
88549  Branch River Air Service Inc.                   0.0   2022  
88565                  TRADEWIND LLC                   0.0   2022  
88600                 Air Trek North                   0.0   2022  
88615                  DIMOND JOAN M                   0.0   2022

In [5]:
#cleaning part
df.drop_duplicates(inplace = True)
#df.isna().sum()
df.dropna(inplace = True)


In [7]:
df["Years"] = df["Event.Date"].dt.year
df.drop(columns = "Event.Date").head(5)

Investigation.Type Accident.Number Aircraft.damage Aircraft.Category  \
22           Accident      ATL82DA027     Substantial        Helicopter   
33           Accident      LAX82FA038       Destroyed          Airplane   
39           Accident      DCA82AA009       Destroyed          Airplane   
43           Accident      LAX82DA037     Substantial          Airplane   
44           Accident      ANC82FA012     Substantial          Airplane   

    Number.of.Engines    Engine.Type                     Air.carrier  \
22                1.0    Turbo Shaft  Rocky Mountain Helicopters, In   
33                2.0  Reciprocating                Lang Air Service   
39                2.0  Reciprocating                 Empire Airlines   
43                1.0  Reciprocating    Joel Frederick's Monarch Air   
44                1.0  Reciprocating       Executive Charter Service   

    Total.Fatal.Injuries  Years  
22                   0.0   1982  
33                   2.0   1982  
39                   2.0   1982  
43                   0.0   1982  
44                   0.0   1982

In [51]:
TotalrecordsByAircraft = df["Aircraft.Category"].value_counts()
pd.DataFrame(TotalrecordsByAircraft)


Aircraft.Category
Airplane                        8670
Helicopter                      1514
Weight-Shift                      60
Powered Parachute                 36
Gyrocraft                         34
Glider                            28
Ultralight                         6
Balloon                            5
Rocket                             1
Blimp                              1

In [13]:
accidentsByAircraft = df[df["Investigation.Type"]=="Accident"].loc[:,["Investigation.Type","Aircraft.Category"]].groupby(["Aircraft.Category","Investigation.Type"]).agg(NumOfAcc = ("Investigation.Type", "count")).sort_values(by = ["NumOfAcc"],ascending = False)
accidentsByAircraft

#groupby the investigation type and aircraft category for accident records only

NumOfAcc
Aircraft.Category Investigation.Type          
Airplane          Accident                8402
Helicopter        Accident                1479
Weight-Shift      Accident                  60
Powered Parachute Accident                  36
Gyrocraft         Accident                  34
Glider            Accident                  28
Ultralight        Accident                   6
Balloon           Accident                   5
Blimp             Accident                   1
Rocket            Accident                   1

## Remarks

On a subset of over 10355 records, 81.14% concerns airplane's accidents, 14.28% Helicopter's, 0.58% Weight-Shift's , 0.35 % Powered Parachute's , 3.3% Gyrocraft , and 0.27% Glider's. The others being irrelevant because of the small records they hold.


In [25]:
TypeAccidentCount = df[df["Investigation.Type"]=="Accident"].loc[:,["Aircraft.Category","Aircraft.damage"]].groupby(["Aircraft.Category","Aircraft.damage"]).agg(TypeOfAccCount = ("Aircraft.damage", "count")).sort_values(by = ["TypeOfAccCount"],ascending = False)
TypeAccidentCount

TypeOfAccCount
Aircraft.Category Aircraft.damage                
Airplane          Substantial                7638
Helicopter        Substantial                1317
Airplane          Destroyed                   696
Helicopter        Destroyed                   158
Airplane          Minor                        66
Weight-Shift      Substantial                  56
Powered Parachute Substantial                  32
Gyrocraft         Substantial                  31
Glider            Substantial                  27
Balloon           Substantial                   5
Ultralight        Substantial                   5
Helicopter        Minor                         4
Gyrocraft         Destroyed                     3
Powered Parachute Destroyed                     3
Weight-Shift      Destroyed                     3
Airplane          Unknown                       2
Glider            Destroyed                     1
Blimp             Destroyed                     1
Powered Parachute Minor                         1
Rocket            Destroyed                     1
Ultralight        Destroyed                     1
Weight-Shift      Minor                         1

<hr>

🔎 Airplane_Destroyed_Rate =   **8.28%** <br>

🔎 Helicopter_Destroyed_Rate =  **10.68 %** ❓ <br> 

🔎 Weight-Shift_Destroyed_Damage_Rate =  **5%** <br>

🔎 Powered Parachute_Destroyed_Rate =  **8.34%** <br>

🔎 Gyrocraft_Destroyed_Rate =  **8.82%** <br>

🔎 Glider_Destroyed_Rate =   **3.57%** <br>

In [32]:
#total fatal injuries by aircraft category and number of engines
Injuries = df[df["Investigation.Type"]=="Accident"].loc[:,["Number.of.Engines","Aircraft.Category","Accident.Number"]].groupby(["Number.of.Engines","Aircraft.Category"]).agg(AccidentCount = ("Accident.Number", "count")).sort_values(by = ["AccidentCount"],ascending = False)
Injuries

AccidentCount
Number.of.Engines Aircraft.Category               
1.0               Airplane                    7176
                  Helicopter                  1370
2.0               Airplane                    1189
                  Helicopter                   109
1.0               Weight-Shift                  60
                  Powered Parachute             36
                  Gyrocraft                     34
                  Glider                        22
4.0               Airplane                      22
3.0               Airplane                      13
1.0               Ultralight                     5
0.0               Glider                         5
1.0               Balloon                        3
8.0               Airplane                       1
2.0               Ultralight                     1
0.0               Airplane                       1
2.0               Glider                         1
                  Balloon                        1
1.0               Rocket                         1
                  Blimp                          1
8.0               Balloon                        1

📍_On the 8402 Airplane accidents_ , most of them are from 1-Engine planes, and we can see(👀) that **0-Engine** and **8-Engine** **Airplane** <br>   only caused **1 accident each !** <br>

📌  We can Also see that across all the Aircrafts, 1-Engine type is the worst leading either most accidents or All 😦!


In [46]:
TotalFatalinjuries = df[df["Investigation.Type"]=="Accident"].loc[:,["Aircraft.Category","Total.Fatal.Injuries"]].groupby(["Aircraft.Category"]).agg(TotalFatalInjuries = ("Total.Fatal.Injuries", "sum")).sort_values(by = ["TotalFatalInjuries"],ascending = False)
TotalFatalinjuries

TotalFatalInjuries
Aircraft.Category                    
Airplane                       3137.0
Helicopter                      578.0
Weight-Shift                     28.0
Gyrocraft                         9.0
Powered Parachute                 6.0
Glider                            4.0
Ultralight                        2.0
Rocket                            1.0
Balloon                           0.0
Blimp                             0.0

In [47]:
TotalFatalinjuries["TotalFatalInjuries"].sum() == df["Total.Fatal.Injuries"].sum()
#This verifies that the sum of Injuries partitioned by each Aircraft adds up to all the ungrouped Injuries  in The original Table.

True

              
              ☠️📉-Death_rate_by_accident = (Total.Fatal.Injuries/Total Accidents) 
              
            
 
 * Airplane_Death_Rate_by_Accident = 37.34% <br>

 * Helicopter_Death_Rate_by_Accident = 39 % <br>

 * Weight-Shift_Death_Rate_by_Accident = 46.67% ❓ <br>

 * Powered Parachute_Death_Rate_by_Accident = 25% <br>

 * Gyrocraft_Death_Rate_by_Accident = 17% <br>

 * Glider_Death_Rate_by_Accident = 14.28% <br>